fe/fe-core/src/main/java/com/starrocks/analysis/DateLiteral.java

```diff
@@ -67,11 +69,16 @@ public class DateLiteral extends LiteralExpr {
     private static final DateTimeFormatter DATE_FORMATTER_TWO_DIGIT;
 
     static {
-        DATE_TIME_FORMATTER = DateUtils.unixDatetimeFormatBuilder("%Y-%m-%e %H:%i:%s").toFormatter();
-        DATE_FORMATTER = DateUtils.unixDatetimeFormatBuilder("%Y-%m-%e").toFormatter();
-        DATE_TIME_FORMATTER_TWO_DIGIT = DateUtils.unixDatetimeFormatBuilder("%y-%m-%e %H:%i:%s").toFormatter();
-        DATE_FORMATTER_TWO_DIGIT = DateUtils.unixDatetimeFormatBuilder("%y-%m-%e").toFormatter();
-        DATE_NO_SPLIT_FORMATTER = DateUtils.unixDatetimeFormatBuilder("%Y%m%e").toFormatter();
+        DATE_TIME_FORMATTER = DateUtils.unixDatetimeFormatBuilder("%Y-%m-%e %H:%i:%s")
```

Comment:
Set `STRICT`  mode in DateUtils.unixDatetimeFormatBuilder method

Reply:
@Seaven  This can not do in `DateUtils.unixDatetimeFormatBuilder`, because setting resolve style can only be deal with after `toFormatter`.
`toFormatter`'s resolve style is `SMART`,  or maybe `SMART` mode is just OK?

Reply:
I think we don't need `SMART` mode anywhere

Reply:
Maybe we can return `Fomatter` directly, not `Builder`